### PR TITLE
Bump version to 0.0.10.0-rc1

### LINKED ISSUE #117: Bump Omni Core version to RC1
Before releasing the release candidate, the version should be bumped to reflect the status.

 - [configure.ac#L13](https://github.com/OmniLayer/omnicore/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/configure.ac#L13)
 - [src/omnicore/version.h#L27](https://github.com/OmniLayer/omnicore/blob/30fdde4833ac20c130d62e9d560b3b0d70ee8744/src/omnicore/version.h#L27)

I think we should go with lower caps `"rc1"`. Do you agree?

It would show up as version string `"omnicore-0.0.10.0-rc1"` and the following file names are expected after going through the Gitian build route (if I'm not mistaken):

 - `omnicore-0.0.10.0-rc1-linux32.tar.gz`
 - `omnicore-0.0.10.0-rc1-linux64.tar.gz`
 - `omnicore-0.0.10.0-rc1-osx-unsigned.tar.gz`
 - `omnicore-0.0.10.0-rc1-win32-setup.exe`
 - `omnicore-0.0.10.0-rc1-win32.zip`
 - `omnicore-0.0.10.0-rc1-win64-setup.exe`
 - `omnicore-0.0.10.0-rc1-win64.zip`
 - `omnicore-0.0.10.0-rc1.tar.gz`

@msgilligan: do you have any experience with software signing? Do you have an Apple Developer ID by any chance?

### PR DESCRIPTION
**Pull request by dexX7**
The version schema was slightly changed, to ensure the resulting number is higher than the version number of Master Core v0.0.9.3.

`OMNICORE_VERSION_BUILD` pretty much serves as dummy/filler in this context.

The version number, used by the alert/notification system:

 - `1000000`

The version string returned by "omni_getinfo":

 - `"0.0.10-rc1"`

This results in the following file names:

 - `omnicore-0.0.10.0-rc1-linux32.tar.gz`
 - `omnicore-0.0.10.0-rc1-linux64.tar.gz`
 - `omnicore-0.0.10.0-rc1-osx-unsigned.tar.gz`
 - `omnicore-0.0.10.0-rc1-win32-setup.exe`
 - `omnicore-0.0.10.0-rc1-win32.zip`
 - `omnicore-0.0.10.0-rc1-win64-setup.exe`
 - `omnicore-0.0.10.0-rc1-win64.zip`
 - `omnicore-0.0.10.0-rc1.tar.gz`

The PR resolves #117, and should be held back until the RC1 is really ready.

### CODE REVIEW

**Comment by dexX7** [comment]
Ping @zathras-crypto.

File: configure.ac [hunk 1]
```diff
@@ -4,13 +4,14 @@ define(_CLIENT_VERSION_MAJOR, 0)
 define(_CLIENT_VERSION_MINOR, 10)
 define(_CLIENT_VERSION_REVISION, 2)
 define(_CLIENT_VERSION_BUILD, 0)
-define(_CLIENT_VERSION_IS_RELEASE, false)
+define(_CLIENT_VERSION_IS_RELEASE, true)
```
Comment (zathras-crypto) [inline configure.ac]: Do release candidates qualify as a release do you think?

**Comment by zathras-crypto** [comment]
> Ping @zathras-crypto.

File: configure.ac (hunk 1 above)
Comment (dexX7) [inline configure.ac]: Hmm.. I think so, yes, but I'm absolutely open to this.

The underlying idea is basically that the RC is as close to the actual release as possible, and if no new issues surface, then RC == release (except the version tag).

On the other hand, if there are indeed issues with the RC, then a version without warning may create the impression it's final.

**Pull request merged**